timeout_sampler/__init__.py

```diff
@@ -228,3 +231,30 @@ def remaining_time(self) -> float:
         Return the remaining part of timeout since the object was created.
         """
         return self.start_time + self.timeout - time.time()
+
+
+def timeout_sampler_deco(
```

Comment:
It would have been nice to give it a nice short catchy and accurate name.
E.g. `waitfor`, `retry`, `until`, etc...

It also deserve some explanation about its behavior in relation to the decorated function.
I.e. that it keeps retrying to call the function until the timeout expires or it successfully returns a `val` that `bool(val)`  resolved to `True`. Exceptions are retried, unless `exceptions_dict` is specified and the raised exception does not belong to that collection.